pkg/apis/flux/v1alpha1/validation/fluxconfig.go

```diff
@@ -122,3 +123,26 @@ func ValidateKustomization(kustomization *fluxv1alpha1.Kustomization, fldPath *f
 
 	return allErrs
 }
+
+// ValidateAdditionalSecretResources validates additionalResources
+func ValidateAdditionalSecretResources(additionalResources []fluxv1alpha1.AdditionalResource, shoot *gardencorev1beta1.Shoot, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if len(additionalResources) == 0 {
+		return allErrs
+	}
+	resources := map[string]gardencorev1beta1.NamedResourceReference{}
+	for _, resource := range shoot.Spec.Resources {
+		resources[resource.Name] = resource
+	}
+	for i, r := range additionalResources {
+		if _, found := resources[r.Name]; !found {
+			allErrs = append(allErrs, field.Invalid(fldPath.Index(i), r.Name, "additionalSecretResource name does not match any of the resource names in Shoot.spec.resources[].name"))
```

Comment:
Should this have the `.name` path?
```suggestion
			allErrs = append(allErrs, field.Invalid(fldPath.Index(i).Child("name"), r.Name, "additionalSecretResource name does not match any of the resource names in Shoot.spec.resources[].name"))
```

Reply:
I guess having the `.name` is slightly more correct.